Sources/StreamChat/StateLayer/Chat.swift

```diff
@@ -170,6 +182,70 @@ public final class Chat {
+    /// It is also possible to create separate ``MemberList`` objects if needed with different filtering options. See ``ChatClient.makeMemberList(with:)``.
+    ///
+    /// - Parameter pagination: The pagination configuration which includes a limit and an offset or a cursor.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    /// - Returns: An array of paginated channel members for the query.
+    @discardableResult public func loadMembers(with pagination: Pagination) async throws -> [ChatChannelMember] {
+        try await memberList.loadMembers(with: pagination)
+    }
+    
+    // MARK: - Member Moderation
+    
+    /// Bans the specified member from the channel.
+    ///
+    /// When the channel member is banned, they will not be allowed to post messages until the ban is removed
+    /// or expired.
+    ///
+    /// Learn more about [banning and moderation tools](https://getstream.io/chat/docs/ios-swift/moderation/?language=swift#ban).
+    ///
+    ///  - Note: Channel watchers cannot be banned.
+    ///  - Note: In most cases, only admins or moderators are allowed to ban other users from the channel.
+    ///
+    /// - Parameters:
+    ///   - userId: The user id of the channel member.
+    ///   - reason: The reason that the ban was created.
+    ///   - timeoutInMinutes: The number of minutes the user should be banned for. Nil means that the user is banned forever or until the user is unbanned explicitly.
+    ///
+    /// - Throws: An error while communicating with the Stream API or missing required capabilities.
+    public func banMember(_ userId: UserId, reason: String? = nil, timeout timeoutInMinutes: Int? = nil) async throws {
+        try state.channel?.requireCapability(of: .banChannelMembers)
+        try await memberUpdater.banMember(userId, in: cid, shadow: false, for: timeoutInMinutes, reason: reason)
+    }
+    
+    /// Shadow bans the specified member from the channel.
+    ///
+    /// When the channel member is shadow banned, they will still be allowed to post messages, but any message
+    /// sent during the ban will only be visible to the author of the message and invisible to other users of the app.
+    ///
+    /// Learn more about [shadow banning and moderation tools](https://getstream.io/chat/docs/ios-swift/moderation/?language=swift#shadow-ban).
+    ///
+    /// - Parameters:
+    ///   - userId: The user id of the channel member.
+    ///   - reason: The reason that the ban was created.
+    ///   - timeoutInMinutes: The number of minutes the user should be banned for. Nil means that the user is banned forever or until the user is unbanned explicitly.
+    ///
+    /// - Throws: An error while communicating with the Stream API or missing required capabilities.
+    public func shadowBanMember(_ userId: UserId, reason: String? = nil, timeout timeoutInMinutes: Int? = nil) async throws {
+        try state.channel?.requireCapability(of: .banChannelMembers)
```

Comment:
Again, these require capibility should not be part of the actual request, but meant to be used in the UI only

Reply:
Removed

Reply:
just seeing this. @nuno-vieira why do we need this only in the UI? We know that we don't have the capability, so we can just throw the error. If we don't, we put too much responsibility on the caller - which doesn't necessarily are our UI components, and customers can forget to do the capability check.

Reply:
yes, just write it down as an open issue and we will discuss is on Thursday, feel free to merge now.

Reply:
The thing is that if the customer forgets to check the capability, the server will Error either way, so it is kind the same thing, although we save 1 request 🤔 The reason I thought that it would be better to skip this check in the LLC is because of the following pros/cons:

Pros of checking capibility in the LLC:
- 1 less request
Cons of checking capibility in the LLC:
1- Maintanance (We need to do this for every action)
2- Error-prone (We can forget to add this for some othe actions)
3- There could be a scenario where the capiblity is updated on the server, and we might block a valid request

Reply:
Added it as a discussion topic for our next meeting

Reply:
Discussion result: let's not check capabilities